activerecord/lib/active_record/store.rb

```diff
@@ -136,6 +166,36 @@ def store_accessor_for(store_attribute)
         type_for_attribute(store_attribute.to_s).accessor
       end
 
+      class AccessorType # :nodoc
+        def self.type_for_accessor(type_name)
+          case type_name
+          when Class
+            type_name.new
+          when Symbol
+            klass = standard_types.fetch(type_name, Type::Value)
+            klass.new
+          else
+            type_name
+          end
+        end
+
+        private
+
+        def self.standard_types
```

Comment:
> Kind of another layer of abstraction.

Yes. Agree, but this doesn't mean we can't use the same concept of type casting applied on `attribute` for this feature too.